airbyte-cdk/python/airbyte_cdk/sources/streams/core.py

```diff
@@ -107,39 +107,26 @@ def get_error_display_message(self, exception: BaseException) -> Optional[str]:
         """
         return None
 
-    def read_full_refresh(
+    def read(  # type: ignore  # ignoring typing for ConnectorStateManager because of circular dependencies
         self,
-        cursor_field: Optional[List[str]],
-        logger: logging.Logger,
-        slice_logger: SliceLogger,
-    ) -> Iterable[StreamData]:
-        slices = self.stream_slices(sync_mode=SyncMode.full_refresh, cursor_field=cursor_field)
-        logger.debug(f"Processing stream slices for {self.name} (sync_mode: full_refresh)", extra={"stream_slices": slices})
-        for _slice in slices:
-            if slice_logger.should_log_slice_message(logger):
-                yield slice_logger.create_slice_log_message(_slice)
-            yield from self.read_records(
-                stream_slice=_slice,
-                sync_mode=SyncMode.full_refresh,
-                cursor_field=cursor_field,
-            )
-
-    def read_incremental(  # type: ignore  # ignoring typing for ConnectorStateManager because of circular dependencies
-        self,
-        cursor_field: Optional[List[str]],
+        # cursor_field: Optional[List[str]],
+        # sync_mode: SyncMode,
+        configured_stream: ConfiguredAirbyteStream,
         logger: logging.Logger,
         slice_logger: SliceLogger,
         stream_state: MutableMapping[str, Any],
         state_manager,
-        per_stream_state_enabled: bool,
         internal_config: InternalConfig,
     ) -> Iterable[StreamData]:
+        sync_mode = configured_stream.sync_mode
+        cursor_field = configured_stream.cursor_field
+
         slices = self.stream_slices(
             cursor_field=cursor_field,
-            sync_mode=SyncMode.incremental,
+            sync_mode=sync_mode,  # todo: change this interface to no longer rely on sync_mode for behavior
```

Comment:
Leaving the todo was intentional. I don't think the `stream_slice()` or `read_records()` should be designed to rely on a `sync_mode`, but I want to try to reduce the complexity and breaking changes in this one PR.

That being said, didn't see any usage of `sync_mode` for internal implementations this in some connectors I glanced at/